extensions/kafka_consumer.go

```diff
@@ -246,8 +247,9 @@ func (q *KafkaConsumer) ConsumeLoop(ctx context.Context) error {
 
 			_, err = q.Consumer.CommitMessage(message)
 			if err != nil {
-				q.handleError(err)
-				return fmt.Errorf("error committing message: %s", err.Error())
+				l.WithError(err).
```

Comment:
What happens if we fail to commit and still process the messages? Couldn't this lead to us reprocessing the message again?

Wouldn't a retry mechanism be better suited to recover from a commit failure?

Reply:
What if we should go with a Committing Before Processing (At-Most-Once Semantics)? The downside is potentially leaving a data loss during processing - but in this case, I'd say we should later implement a dead-letter queue to retry

Reply:
Well, in hindsight I think the approach depends on what is more critical: not delivering a message or delivering more than once. Maybe I'm biased from the last incident of consuming multiple times and not delivering the message is more critical than delivering more than once.

WDYT?